src/otaclient/boot_control/_grub.py

```diff
@@ -838,108 +837,32 @@ def _copy_boot_files_from_standby_slot(self):
-            self._update_fstab(
-                standby_slot_fstab=standby_fstab,
-                active_slot_fstab=active_fstab,
-            )
+    def _pre_update_platform_specific(
+        self, *, standby_as_ref: bool, erase_standby: bool
+    ) -> None:
+        """GRUB-specific pre-update: cleanup standby ota_partition folder."""
+        # remove old files under standby ota_partition folder
+        self._cleanup_standby_ota_partition_folder()
+
+    def _post_update_platform_specific(self, *, update_version: str) -> None:
+        """GRUB-specific post-update: update fstab, copy boot files, and reboot to standby."""
+        # ------ update fstab ------ #
+        active_fstab = self._mp_control.active_slot_mount_point / Path(
+            boot_cfg.FSTAB_FILE_PATH
+        ).relative_to("/")
+        standby_fstab = self._mp_control.standby_slot_mount_point / Path(
+            boot_cfg.FSTAB_FILE_PATH
+        ).relative_to("/")
+        self._update_fstab(
+            standby_slot_fstab=standby_fstab,
+            active_slot_fstab=active_fstab,
+        )
 
-            # ------ prepare boot files ------ #
-            self._copy_boot_files_from_standby_slot()
+        # ------ prepare boot files ------ #
+        self._copy_boot_files_from_standby_slot()
 
-            # ------ pre-reboot ------ #
-            self._mp_control.umount_all(ignore_error=True)
-            self._boot_control.grub_reboot_to_standby()
-        except Exception as e:
-            _err_msg = f"failed on post_update: {e!r}"
-            logger.error(_err_msg)
-            raise ota_errors.BootControlPostUpdateFailed(
-                _err_msg, module=__name__
-            ) from e
+        # ------ configure boot to standby ------ #
+        self._boot_control.grub_reboot_to_standby()
 
     def finalizing_update(self, *, chroot: str | None = None) -> NoReturn:
         cmdhelper.reboot(chroot=chroot)
-
-    def pre_rollback(self):
-        try:
-            logger.info("grub_boot: pre-rollback setup...")
-            self._ota_status_control.pre_rollback_current()
-            self._mp_control.mount_standby()
-        except Exception as e:
-            _err_msg = f"failed on pre_rollback: {e!r}"
-            logger.error(_err_msg)
-            raise ota_errors.BootControlPreRollbackFailed(
-                _err_msg, module=__name__
-            ) from e
-
-    def post_rollback(self):
-        try:
-            logger.info("grub_boot: post-rollback setup...")
-            self._ota_status_control.post_rollback_standby()
-            self._boot_control.grub_reboot_to_standby()
-            self._mp_control.umount_all(ignore_error=True)
-        except Exception as e:
-            _err_msg = f"failed on pre_rollback: {e!r}"
-            logger.error(_err_msg)
-            raise ota_errors.BootControlPostRollbackFailed(
-                _err_msg, module=__name__
-            ) from e
-
-    finalizing_rollback = finalizing_update
```

Comment:
Moved to abstraction class(`_base.py`) and keep the bootloader-specific controls.